lib/Sema/CSSimplify.cpp

```diff
@@ -9033,31 +9034,7 @@ ConstraintSystem::SolutionKind
 ConstraintSystem::simplifyOpaqueUnderlyingTypeConstraint(Type type1, Type type2,
                                              TypeMatchOptions flags,
                                              ConstraintLocatorBuilder locator) {
-  // Open the second type, which must be an opaque archetype, to try to
-  // infer the first type using its constraints.
-  auto opaque2 = type2->castTo<OpaqueTypeArchetypeType>();
-
-  // Open the generic signature of the opaque decl, and bind the "outer" generic
-  // params to our context. The remaining axes of freedom on the type variable
-  // corresponding to the underlying type should be the constraints on the
-  // underlying return type.
-  OpenedTypeMap replacements;
-  openGeneric(DC, opaque2->getBoundSignature(), locator, replacements);
-
-  auto underlyingTyVar = openType(opaque2->getInterfaceType(),
-                                  replacements);
-  assert(underlyingTyVar);
-  
-  if (auto dcSig = DC->getGenericSignatureOfContext()) {
-    for (auto param : dcSig->getGenericParams()) {
-      addConstraint(ConstraintKind::Bind,
-                    openType(param, replacements),
-                    DC->mapTypeIntoContext(param),
-                    locator);
-    }
-  }
-
-  addConstraint(ConstraintKind::Equal, type1, underlyingTyVar, locator);
+  addConstraint(ConstraintKind::Equal, type1, type2, locator);
```

Comment:
I think `OpaqueUnderlyingType` constraint could be removed all together at this point since it became just a regular equality without any special semantics.